com.civicrm.hrjobroles/CRM/Hrjobroles/BAO/HrJobRoles.php

```diff
@@ -47,21 +47,21 @@ public static function getContactRoles($cuid)  {
   }
 
   /**
-   * Get options for a given job roles field along with their database IDs.
+   * Get option values for specific option group.
    *
    * @param String $fieldName
    *
-   * @return Array
+   * @return array
    */
   public static function buildDbOptions($fieldName) {
     $queryParam = array(1 => array($fieldName, 'String'));
-    $query = "SELECT cpv.id, cpv.label from civicrm_option_value cpv
+    $query = "SELECT cpv.value, cpv.label from civicrm_option_value cpv
               LEFT JOIN civicrm_option_group cpg on cpv.option_group_id = cpg.id
               WHERE cpg.name = %1";
     $options = array();
     $result = CRM_Core_DAO::executeQuery($query, $queryParam);
     while ($result->fetch()) {
-      $options[] =  array( 'id'=>$result->id, 'label'=>strtolower($result->label) );
+      $options[] =  array( 'value'=>$result->value, 'label'=>strtolower($result->label) );
     }
     return $options;
   }
```

Comment:
Looking at CRM_Core_BAO_OptionValue::add() I can see it calls CRM_Core_PseudoConstant::flush() to clear the cache, but it clears ALL the cache, and that might be why it is taking so long. I also see that the flush method accepts a pseudconstant name as parameter and, in that case, it will only clear this pseudoconstant values. Maybe that would be faster?